packages/generation/src/generation/recipe/gan_recipe.py

```diff
@@ -56,14 +58,15 @@ def training_step(
 
             # discriminator
             discriminator_output: dict[str, list[Tensor]] = self.model.discriminator(
-                clean_mix, generated_mix.detach()
+                clean_mix, generated_mix
             )
 
             discriminator_loss = self.loss.discriminator(discriminator_output)
 
-            discriminator_optimizer.zero_grad()
-            self.manual_backward(discriminator_loss["loss"], retain_graph=True)
-            discriminator_optimizer.step()
+            if self.global_step >= self.num_frozen_steps:
```

Comment:
I was looking at the loss graphs, and they didn't seem quite right, so I went back and compared this to how it's done in bigvgan:

https://github.com/NVIDIA/BigVGAN/blob/main/train.py#L486

I their implementation (different than here), the `discriminator_optimizer.zero_grad()` is _always_ called, but backwards and step are optionally called. 

I think the way you have it here we may be implicitly stepping the discriminator through the generator because it's not zeroing the gradients.